modules/exploits/multi/http/cve_2014_9390.rb

```diff
@@ -0,0 +1,378 @@
+
+        On operating systems which have case-insensitive file systems, like
+        Windows and OS X, Git clients can be convinced to retrieve and
+        overwrite sensitive configuration files in the .git
+        directory which can allow arbitrary code execution if a vulnerable
+        client can be convinced to perform certain actions (for example,
+        a checkout) against a malicious Git repository.
+
+        A second vulnerability with similar characteristics also exists in both
+        Git and Mercurial clients, on HFS+ file systems (Mac OS X) only, where
+        certain Unicode codepoints are ignorable.
+
+        The third vulnerability with similar characteristics only affects
+        Mercurial clients on Windows, where Windows "short names"
+        (MS-DOS-compatible 8.3 format) are supported.
+
+        Today this module only truly supports the first vulnerability (Git
+        clients on case-insensitive file systems) but has the functionality to
+        support the remaining two with a little work.
+      ),
+      'License' => MSF_LICENSE,
+      'Author' => [
+        'Jon Hart <jon_hart[at]rapid7.com>' # metasploit module
+      ],
+      'References'     =>
+        [
+          ['CVE', '2014-9390'],
+          ['URL', 'http://git-blame.blogspot.com.es/2014/12/git-1856-195-205-214-and-221-and.html'],
+          ['URL', 'http://article.gmane.org/gmane.linux.kernel/1853266'],
+          ['URL', 'https://community.rapid7.com/community/metasploit/blog/2014/12/30/12-days-of-haxmas-exploiting-cve-2014-9390-in-git-and-mercurial'],
+          ['URL', 'https://github.com/blog/1938-vulnerability-announced-update-your-git-clients'],
+          ['URL', 'https://www.mehmetince.net/one-git-command-may-cause-you-hacked-cve-2014-9390-exploitation-for-shell/'],
+          ['URL', 'http://mercurial.selenic.com/wiki/WhatsNew#Mercurial_3.2.3_.282014-12-18.29'],
+          ['URL', 'http://selenic.com/repo/hg-stable/rev/c02a05cc6f5e'],
+          ['URL', 'http://selenic.com/repo/hg-stable/rev/6dad422ecc5a']
+
+        ],
+      'DisclosureDate' => 'Dec 18 2014',
+      'Targets' =>
+        [
+          [
+            'Automatic',
+            {
+              'Platform' => [ 'unix' ],
+              'Arch'     => ARCH_CMD,
+              'Payload'        =>
+                {
+                  'Compat'      =>
+                    {
+                      'PayloadType' => 'cmd cmd_bash',
+                      'RequiredCmd' => 'generic bash-tcp perl bash'
+                    }
+                }
+            }
+          ],
+          [
+            'Windows Powershell',
+            {
+              'Platform' => [ 'windows' ],
+              'Arch'     => [ARCH_X86, ARCH_X86_64]
+            }
+          ]
+        ],
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptBool.new('GIT', [true, 'Exploit Git clients', true]),
+        OptBool.new('MERCURIAL', [true, 'Exploit Mercurial clients', false])
```

Comment:
The Mercurial option really shouldn't be present at all; it's just going to build up the hopes of the user. I don't mind the rest of the detail and the call to action to help, but it seems like the Mercurial option doesn't do anything other than frustrate the user when it doesn't work today.

At any rate, the boolean shouldn't be required. I'll fix this upon landing, and be explicit about not enabling it. Something like

```ruby
OptBool.new('MERCURIAL', [false, 'Enable experimental Mercurial functionality', false])
```

seems more appropriate today.